servicetalk-concurrent-api/src/test/java/io/servicetalk/concurrent/api/publisher/TimeoutPublisherTest.java

```diff
@@ -348,6 +350,19 @@ public Cancellable execute(final Runnable task) throws RejectedExecutionExceptio
         assertThat(subscriber.awaitOnError(), instanceOf(TimeoutException.class));
     }
 
+    @Test
+    @RepeatedTest(100)
+    void timeoutDemandTimerCancellation() {
+        init(TimerBehaviorParam.DEMAND_TIMER, Duration.ofSeconds(1));
+        Subscription subscription = subscriber.awaitSubscription();
+        subscription.request(1);
+        assertThat(testExecutor.scheduledTasksPending(), is(0));
+        ForkJoinPool.commonPool().execute(() -> publisher.onNext(1));
```

Comment:
I'm going to guess this isn't the right thread pool to use. I see other tests use a different `ExecutionExtension` but we already have one of those that returns the `TextExecutor` and we need a real thread pool to induce the race.
Suggestions welcome.

Reply:
We typically create/destroy an executor in the test if required:

```java
    @BeforeAll
    static void beforeClass() {
        jdkExecutor = java.util.concurrent.Executors.newCachedThreadPool();
    }

    @AfterAll
    static void afterClass() {
        if (jdkExecutor != null) {
            jdkExecutor.shutdown();
        }
    }
```